src/utils/extractBrowserDetails.js

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+export const extractBrowserDetails = () => {
```

Comment:
Agree with @jasonwaters, especially because the context creation needs to be forwards-compatible with User Agent Hints API. The User Agent string will eventually be removed. `getBrowser.js` will need to be updated eventually, so it would be easier to only have to update one place instead of two.